lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -589,6 +639,36 @@ static LogicalResult verifyEnableOp(EnableOp enableOp) {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// IfOp
+//===----------------------------------------------------------------------===//
+
+static LogicalResult verifyIfOp(IfOp ifOp) {
+  auto component = ifOp->getParentOfType<ComponentOp>();
+  auto wiresOp = component.getWiresOp();
+  auto groupName = ifOp.groupName();
+  auto groupOp = wiresOp.lookupSymbol<GroupOp>(groupName);
+
+  if (!groupOp)
+    return ifOp.emitOpError()
+           << "with group '" << groupName << "', which does not exist.";
+
+  if (ifOp.thenRegion().front().empty())
+    return ifOp.emitError() << "empty 'then' region.";
+
+  if (ifOp.elseRegion().getBlocks().size() != 0 &&
+      ifOp.elseRegion().front().empty())
+    return ifOp.emitError() << "empty 'else' region.";
+
+  if (failed(isPortDrivenByGroup(ifOp.cond(), groupOp)))
+    return ifOp.emitError()
+           << "conditional op: '" << valueName(component, ifOp.cond())
```

Comment:
Awesome error message.

Reply:
So when I said `conditional op` I meant we should link directly to the `op` for which this port is related to.

E.g.
```
%c0.in, %c0.out = "c0" @MyComp
// c0 is the op
// c0.in, c0.out are the ports.
```

If you like the above better, changing it back to `conditional port` is fine. Sorry for any confusion.